### ISSUE TITLE
Update DistributeService example with Akka HTTP

### ISSUE THREAD

**Issue by manuzhang** [gearpump/gearpump#2107]
The current DistributeService example uses spray which has been deprecated by Akka HTTP

**Comment by domsj** [comment]
I tried importing the repo in intellij and was greeted with the following error

```
Error while importing sbt project:

[info] Loading global plugins from /home/jan/.sbt/1.0/plugins
[info] Loading settings for project gearpump-build from plugins.sbt ...
[info] Loading project definition from /home/jan/IdeaProjects/gearpump/project
[info] Loading settings for project gearpump from jvm.sbt,version.sbt,build.sbt ...
[info] Set current project to gearpump (in build file:/home/jan/IdeaProjects/gearpump/)
[info] sbt server started at local:///home/jan/.sbt/1.0/server/ed3c1f56d6d6645fe3e3/sock
sbt:gearpump>
[info] Defining Global / sbtStructureOptions, Global / sbtStructureOutputFile and 1 others.
[info] The new values will be used by cleanKeepFiles
[info] 	Run `last` for details.
[info] Reapplying settings...
[info] Set current project to gearpump (in build file:/home/jan/IdeaProjects/gearpump/)
[info] Applying State transformations org.jetbrains.sbt.CreateTasks from /home/jan/.IdeaIC2018.3/config/plugins/Scala/launcher/sbt-structure-1.0.jar
[info] Reapplying settings...
[info] Set current project to gearpump (in build file:/home/jan/IdeaProjects/gearpump/)
[info] Updating gearpump-examples-distributeservice...
[warn] 	module not found: io.spray#spray-can_2.12;1.3.2
[warn] ==== local: tried
[warn]   /home/jan/.ivy2/local/io.spray/spray-can_2.12/1.3.2/ivys/ivy.xml
[warn] ==== public: tried
[warn]   https://repo1.maven.org/maven2/io/spray/spray-can_2.12/1.3.2/spray-can_2.12-1.3.2.pom
[warn] ==== local-preloaded-ivy: tried
[warn]   /home/jan/.sbt/preloaded/io.spray/spray-can_2.12/1.3.2/ivys/ivy.xml
[warn] ==== local-preloaded: tried
[warn]   file:////home/jan/.sbt/preloaded/io/spray/spray-can_2.12/1.3.2/spray-can_2.12-1.3.2.pom
[warn] ==== apache-repo: tried
[warn]   https://repository.apache.org/content/repositories/io/spray/spray-can_2.12/1.3.2/spray-can_2.12-1.3.2.pom
[warn] ==== sonatype-releases: tried
[warn]   https://oss.sonatype.org/content/repositories/releases/io/spray/spray-can_2.12/1.3.2/spray-can_2.12-1.3.2.pom
[warn] 	module not found: io.spray#spray-routing-shapeless2_2.12;1.3.2
[warn] ==== local: tried
[warn]   /home/jan/.ivy2/local/io.spray/spray-routing-shapeless2_2.12/1.3.2/ivys/ivy.xml
[warn] ==== public: tried
[warn]   https://repo1.maven.org/maven2/io/spray/spray-routing-shapeless2_2.12/1.3.2/spray-routing-shapeless2_2.12-1.3.2.pom
[warn] ==== local-preloaded-ivy: tried
[warn]   /home/jan/.sbt/preloaded/io.spray/spray-routing-shapeless2_2.12/1.3.2/ivys/ivy.xml
[warn] ==== local-preloaded: tried
[warn]   file:////home/jan/.sbt/preloaded/io/spray/spray-routing-shapeless2_2.12/1.3.2/spray-routing-shapeless2_2.12-1.3.2.pom
[warn] ==== apache-repo: tried
[warn]   https://repository.apache.org/content/repositories/io/spray/spray-routing-shapeless2_2.12/1.3.2/spray-routing-shapeless2_2.12-1.3.2.pom
[warn] ==== sonatype-releases: tried
[warn]   https://oss.sonatype.org/content/repositories/releases/io/spray/spray-routing-shapeless2_2.12/1.3.2/spray-routing-shapeless2_2.12-1.3.2.pom
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	::          UNRESOLVED DEPENDENCIES         ::
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	:: io.spray#spray-can_2.12;1.3.2: not found
[warn] 	:: io.spray#spray-routing-shapeless2_2.12;1.3.2: not found
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn]
[warn] 	Note: Unresolved dependencies path:
[warn] 		io.spray:spray-can_2.12:1.3.2 (/home/jan/IdeaProjects/gearpump/build.sbt#L189)
[warn] 		  +- io.gearpump:gearpump-examples-distributeservice_2.12:0.9.0
[warn] 		io.spray:spray-routing-shapeless2_2.12:1.3.2 (/home/jan/IdeaProjects/gearpump/build.sbt#L189)
[warn] 		  +- io.gearpump:gearpump-examples-distributeservice_2.12:0.9.0
[error] sbt.librarymanagement.ResolveException: unresolved dependency: io.spray#spray-can_2.12;1.3.2: not found
[error] unresolved dependency: io.spray#spray-routing-shapeless2_2.12;1.3.2: not found
[error] 	at sbt.internal.librarymanagement.IvyActions$.resolveAndRetrieve(IvyActions.scala:332)
[error] 	at sbt.internal.librarymanagement.IvyActions$.$anonfun$updateEither$1(IvyActions.scala:208)
[error] 	at sbt.internal.librarymanagement.IvySbt$Module.$anonfun$withModule$1(Ivy.scala:239)
[error] 	at sbt.internal.librarymanagement.IvySbt.$anonfun$withIvy$1(Ivy.scala:204)
[error] 	at sbt.internal.librarymanagement.IvySbt.sbt$internal$librarymanagement$IvySbt$$action$1(Ivy.scala:70)
[error] 	at sbt.internal.librarymanagement.IvySbt$$anon$3.call(Ivy.scala:77)
[error] 	at xsbt.boot.Locks$GlobalLock.withChannel$1(Locks.scala:95)
[error] 	at xsbt.boot.Locks$GlobalLock.xsbt$boot$Locks$GlobalLock$$withChannelRetries$1(Locks.scala:80)
[error] 	at xsbt.boot.Locks$GlobalLock$$anonfun$withFileLock$1.apply(Locks.scala:99)
[error] 	at xsbt.boot.Using$.withResource(Using.scala:10)
[error] 	at xsbt.boot.Using$.apply(Using.scala:9)
[error] 	at xsbt.boot.Locks$GlobalLock.ignoringDeadlockAvoided(Locks.scala:60)
[error] 	at xsbt.boot.Locks$GlobalLock.withLock(Locks.scala:50)
[error] 	at xsbt.boot.Locks$.apply0(Locks.scala:31)
[error] 	at xsbt.boot.Locks$.apply(Locks.scala:28)
[error] 	at sbt.internal.librarymanagement.IvySbt.withDefaultLogger(Ivy.scala:77)
[error] 	at sbt.internal.librarymanagement.IvySbt.withIvy(Ivy.scala:199)
[error] 	at sbt.internal.librarymanagement.IvySbt.withIvy(Ivy.scala:196)
[error] 	at sbt.internal.librarymanagement.IvySbt$Module.withModule(Ivy.scala:238)
[error] 	at sbt.internal.librarymanagement.IvyActions$.updateEither(IvyActions.scala:193)
[error] 	at sbt.librarymanagement.ivy.IvyDependencyResolution.update(IvyDependencyResolution.scala:20)
[error] 	at sbt.librarymanagement.DependencyResolution.update(DependencyResolution.scala:56)
[error] 	at sbt.internal.LibraryManagement$.resolve$1(LibraryManagement.scala:45)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$12(LibraryManagement.scala:93)
[error] 	at sbt.util.Tracked$.$anonfun$lastOutput$1(Tracked.scala:68)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$19(LibraryManagement.scala:106)
[error] 	at scala.util.control.Exception$Catch.apply(Exception.scala:224)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11(LibraryManagement.scala:106)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11$adapted(LibraryManagement.scala:89)
[error] 	at sbt.util.Tracked$.$anonfun$inputChanged$1(Tracked.scala:149)
[error] 	at sbt.internal.LibraryManagement$.cachedUpdate(LibraryManagement.scala:120)
[error] 	at sbt.Classpaths$.$anonfun$updateTask$5(Defaults.scala:2561)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:44)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:40)
[error] 	at sbt.std.Transform$$anon$4.work(System.scala:67)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:269)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:16)
[error] 	at sbt.Execute.work(Execute.scala:278)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:269)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
[error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error] 	at java.lang.Thread.run(Thread.java:748)
[error] sbt.librarymanagement.ResolveException: unresolved dependency: io.spray#spray-can_2.12;1.3.2: not found
[error] unresolved dependency: io.spray#spray-routing-shapeless2_2.12;1.3.2: not found
[error] 	at sbt.internal.librarymanagement.IvyActions$.resolveAndRetrieve(IvyActions.scala:332)
[error] 	at sbt.internal.librarymanagement.IvyActions$.$anonfun$updateEither$1(IvyActions.scala:208)
[error] 	at sbt.internal.librarymanagement.IvySbt$Module.$anonfun$withModule$1(Ivy.scala:239)
[error] 	at sbt.internal.librarymanagement.IvySbt.$anonfun$withIvy$1(Ivy.scala:204)
[error] 	at sbt.internal.librarymanagement.IvySbt.sbt$internal$librarymanagement$IvySbt$$action$1(Ivy.scala:70)
[error] 	at sbt.internal.librarymanagement.IvySbt$$anon$3.call(Ivy.scala:77)
[error] 	at xsbt.boot.Locks$GlobalLock.withChannel$1(Locks.scala:95)
[error] 	at xsbt.boot.Locks$GlobalLock.xsbt$boot$Locks$GlobalLock$$withChannelRetries$1(Locks.scala:80)
[error] 	at xsbt.boot.Locks$GlobalLock$$anonfun$withFileLock$1.apply(Locks.scala:99)
[error] 	at xsbt.boot.Using$.withResource(Using.scala:10)
[error] 	at xsbt.boot.Using$.apply(Using.scala:9)
[error] 	at xsbt.boot.Locks$GlobalLock.ignoringDeadlockAvoided(Locks.scala:60)
[error] 	at xsbt.boot.Locks$GlobalLock.withLock(Locks.scala:50)
[error] 	at xsbt.boot.Locks$.apply0(Locks.scala:31)
[error] 	at xsbt.boot.Locks$.apply(Locks.scala:28)
[error] 	at sbt.internal.librarymanagement.IvySbt.withDefaultLogger(Ivy.scala:77)
[error] 	at sbt.internal.librarymanagement.IvySbt.withIvy(Ivy.scala:199)
[error] 	at sbt.internal.librarymanagement.IvySbt.withIvy(Ivy.scala:196)
[error] 	at sbt.internal.librarymanagement.IvySbt$Module.withModule(Ivy.scala:238)
[error] 	at sbt.internal.librarymanagement.IvyActions$.updateEither(IvyActions.scala:193)
[error] 	at sbt.librarymanagement.ivy.IvyDependencyResolution.update(IvyDependencyResolution.scala:20)
[error] 	at sbt.librarymanagement.DependencyResolution.update(DependencyResolution.scala:56)
[error] 	at sbt.internal.LibraryManagement$.resolve$1(LibraryManagement.scala:45)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$12(LibraryManagement.scala:93)
[error] 	at sbt.util.Tracked$.$anonfun$lastOutput$1(Tracked.scala:68)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$19(LibraryManagement.scala:106)
[error] 	at scala.util.control.Exception$Catch.apply(Exception.scala:224)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11(LibraryManagement.scala:106)
[error] 	at sbt.internal.LibraryManagement$.$anonfun$cachedUpdate$11$adapted(LibraryManagement.scala:89)
[error] 	at sbt.util.Tracked$.$anonfun$inputChanged$1(Tracked.scala:149)
[error] 	at sbt.internal.LibraryManagement$.cachedUpdate(LibraryManagement.scala:120)
[error] 	at sbt.Classpaths$.$anonfun$updateTask$5(Defaults.scala:2561)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:44)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:40)
[error] 	at sbt.std.Transform$$anon$4.work(System.scala:67)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:269)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:16)
[error] 	at sbt.Execute.work(Execute.scala:278)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:269)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
[error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error] 	at java.lang.Thread.run(Thread.java:748)
[error] (gearpump-examples-distributeservice / update) sbt.librarymanagement.ResolveException: unresolved dependency: io.spray#spray-can_2.12;1.3.2: not found
[error] unresolved dependency: io.spray#spray-routing-shapeless2_2.12;1.3.2: not found
[error] (gearpump-examples-distributeservice / ssExtractDependencies) sbt.librarymanagement.ResolveException: unresolved dependency: io.spray#spray-can_2.12;1.3.2: not found
[error] unresolved dependency: io.spray#spray-routing-shapeless2_2.12;1.3.2: not found
[error] Total time: 8 s, completed Dec 31, 2018 3:07:47 PM
[info] shutting down server
```

which seems to be related to this ticket.

Checking out 0.8.0 resulted in a proper import by intellij.

**Comment by manuzhang** [comment]
Yes, please allow me some time to update the example. Meanwhile you may remove the example locally.

**Comment by domsj** [comment]
ticket can be closed now #2124 was merged? (or do you want to keep the ticket open, to at a later time re-instate the example, using akka-http iso spray ?)

**Comment by manuzhang** [comment]
Yes, it can be close. The use case in the example is not our focus now.
